Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -326,21 +327,78 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f, amrex::Real
                                      "should contain only one species\n");
     std::pair<std::string,openPMD::ParticleSpecies> ps = *i.particles.begin();
 
-    //TODO: In future PRs will add AMREX_ALWAYS_ASSERT_WITH_MESSAGE to test if mass and charge are both const
+    //TODO: In future PRs will add ASSERT_WITH_MESSAGE to test if mass and charge are both const
     amrex::Real p_m = ps.second["mass"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::Real>().get()[0];
     amrex::Real p_q = ps.second["charge"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::Real>().get()[0];
-    int npart = ps.second["position"]["x"].getExtent()[0];
+    long npart = ps.second["position"]["x"].getExtent()[0];
     series.flush();
 
+    //Conversion from Geant4 system of units (http://geant4.web.cern.ch/sites/geant4.web.cern.ch/files/geant4/collaboration/working_groups/electromagnetic/gallery/units/SystemOfUnits.html) to WarpX (SI)
     mass = p_m*PhysConst::mevpc2_kg;
     charge = p_q*PhysConst::q_e;
-    Real const weight = q_tot/(charge*amrex::Real(npart));
+    amrex::Real mmpns_mps = 1.e6;
+    amrex::Real mm_m = 1.e-3;
+    amrex::Real weight;
+    if (physical_q_tot!=0.0){
+        weight = physical_q_tot/(charge*amrex::Real(npart));
+    }
+    else {
+        //In case openPMD file contains macroparticle instead of real particle data
+        weight = charge;
+    }
+
+    std::shared_ptr<amrex::Real> ptr_x = ps.second["position"]["x"].loadChunk<amrex::Real>();
+    series.flush();
+    std::shared_ptr<amrex::Real> ptr_vx = ps.second["velocity"]["x"].loadChunk<amrex::Real>();
+    series.flush();
+    std::shared_ptr<amrex::Real> ptr_z = ps.second["position"]["z"].loadChunk<amrex::Real>();
+    series.flush();
+    std::shared_ptr<amrex::Real> ptr_vz = ps.second["velocity"]["z"].loadChunk<amrex::Real>();
+    series.flush();
+#if (defined WARPX_DIM_3D)
+    std::shared_ptr<amrex::Real> ptr_y = ps.second["position"]["y"].loadChunk<amrex::Real>();
+    series.flush();
+    std::shared_ptr<amrex::Real> ptr_vy = ps.second["velocity"]["y"].loadChunk<amrex::Real>();
+    series.flush();
+#endif
 
-    amrex::Print() << npart << " parts of species " << ps.first << "\nWith"
-    << " mass = " << mass << " and charge = " << charge << "\nTo initialize "
-    << npart << " macroparticles with weight " << weight << "\n";
+    // Allocate temporary vectors on the CPU
```

Comment:
This only declares the variable but does not yet allocate memory:
```suggestion
    // Declare temporary vectors on the CPU
```

Reply:
I didn't know that option. But it makes sense to use it here. I will fix it and thanks!!

Reply:
Forgot to post the reference:
- `std::vector`: https://en.cppreference.com/w/cpp/container/vector/reserve
- AMReX' `HostVector`:
  - https://github.com/AMReX-Codes/amrex/blob/20.04/Src/Base/AMReX_GpuContainers.H#L32-L33 https://github.com/AMReX-Codes/amrex/blob/20.04/Src/Base/AMReX_PODVector.H#L432-L435

Reply:
There is a problem with this idea. It assumes all particles fall inside the simulation box region. I don't want to add that constraint. I added a warning if they were not inside the box. But then I do not wish to reserve the vectors. Is that ok?